lib/mindee/http/endpoint.rb

```diff
@@ -31,14 +31,16 @@ class Endpoint
       attr_reader :api_key
       # @return [Integer]
       attr_reader :request_timeout
+      # @return [String]
+      attr_reader :url_root
 
       def initialize(owner, url_name, version, api_key: '')
         @owner = owner
         @url_name = url_name
         @version = version
         @request_timeout = ENV.fetch(REQUEST_TIMEOUT_ENV_NAME, TIMEOUT_DEFAULT).to_i
         @api_key = api_key.nil? || api_key.empty? ? ENV.fetch(API_KEY_ENV_NAME, API_KEY_DEFAULT) : api_key
-        @url_root = "#{BASE_URL_DEFAULT}/products/#{@owner}/#{@url_name}/v#{@version}"
+        update_url_root(BASE_URL_DEFAULT)
```

Comment:
Why not simplify to something like this?
```suggestion
        base_url = ENV.fetch(BASE_URL_ENV_NAME, BASE_URL_DEFAULT)
        @url_root = "#{base_url}/products/#{@owner}/#{@url_name}/v#{@version}"
```